js/plugins/dotprompt/src/prompt.ts

```diff
@@ -197,16 +198,41 @@ export class Dotprompt<Variables = unknown> implements PromptMetadata {
     return this._generateOptions<O>(opt);
   }
 
+  async renderInNewSpan<O extends z.ZodTypeAny = z.ZodTypeAny>(
+    opt: PromptGenerateOptions<Variables>
+  ): Promise<GenerateOptions<z.ZodTypeAny, O>> {
+    return runInNewSpan(
+      {
+        metadata: {
```

Comment:
Should we include variant info in here, or will that be bundled in the name already?

Reply:
Good point. Variant isn't included in the name by default. I will add that.